src/query/storage/m3/consolidators/multi_fetch_result.go

```diff
@@ -190,11 +202,22 @@ func (r *multiResult) Add(
 		return
 	}
 
+	// the series limit was reached within this namespace.
+	if !metadata.Exhaustive && r.limitOpts.RequireExhaustive {
```

Comment:
this is the case where query series limit was exceeded by a single namespace (i.e summing up the results from all the instances in a replica)